rs/canister_sandbox/src/replica_controller/sandbox_process_eviction.rs

```diff
@@ -1,43 +1,69 @@
 use num_traits::ops::saturating::SaturatingAdd;
 use std::time::Instant;
 
-use ic_types::{CanisterId, NumBytes};
+use ic_types::{AccumulatedPriority, CanisterId, NumBytes};
 
 #[derive(Clone, Eq, PartialEq, Debug)]
 pub(crate) struct EvictionCandidate {
     pub id: CanisterId,
     pub last_used: Instant,
     pub rss: NumBytes,
+    pub scheduler_priority: AccumulatedPriority,
 }
 
-/// Evicts the least recently used candidates in order to bring the number of
-/// the remaining candidates down to `max_count_threshold` and their total RSS
-/// down to `max_sandboxes_rss`.
+/// Evicts the candidates with lowest scheduler priority in order to bring the
+/// number of the remaining candidates down to `max_count_threshold` and their
+/// total RSS down to `max_sandboxes_rss`.
 ///
 /// The function also tries to evict candidates that have been idle for a long
 /// time (`last_used_threshold`).
 pub(crate) fn evict(
-    mut candidates: Vec<EvictionCandidate>,
+    candidates: Vec<EvictionCandidate>,
     total_rss: NumBytes,
     max_count_threshold: usize,
     last_used_threshold: Instant,
     max_sandboxes_rss: NumBytes,
 ) -> Vec<EvictionCandidate> {
-    candidates.sort_by_key(|x| x.last_used);
+    let evict_at_least: usize = candidates.len().saturating_sub(max_count_threshold);
 
-    let evict_at_least = candidates.len().saturating_sub(max_count_threshold);
+    let (mut idle, mut non_idle): (_, Vec<_>) = candidates
+        .into_iter()
+        .partition(|candidate| candidate.last_used < last_used_threshold);
```

Comment:
Nit: This causes a whole lot of copying, whether any sandbox needs to be evicted or not.

At the very least I'd put a check at the top to see if any work needs doing.

As a matter of fact, even just putting together the `candidates` vector (in the caller) is already a huge waste of time. So `evict_sandbox_processes()` should also have a similar check at the top.

Reply:
The code is updated, and the check is implicit inside the first iteration of for loop.

Reply:
That was changed in this commit https://github.com/dfinity/ic/pull/2590/commits/ddb7f78156c0ac26efa95e1b0757297adf0ac531 .

Reply:
I do not believe this PR is the place where we should optimize the creation of the candidates Vec in `evict_sandbox_processes`, that was not the main point here.

Reply:
As you wish.

All it would mean is an extra `if <below limit> { return }`. But it's just a suggestion, not a blocker. And David already approved on behalf of Message Routing.